diff --git a/apple/internal/ios_rules.bzl b/apple/internal/ios_rules.bzl
index 9b0d1270..73dfb375 100644
--- a/apple/internal/ios_rules.bzl
+++ b/apple/internal/ios_rules.bzl
@@ -927,6 +927,7 @@ def _ios_extension_impl(ctx):
         attr = ctx.attr,
         res_attrs = [
             "app_icons",
+            "resources",
             "strings",
         ],
     )
diff --git a/tools/plisttool/plisttool.py b/tools/plisttool/plisttool.py
index ecc84f28..2cd295cc 100644
--- a/tools/plisttool/plisttool.py
+++ b/tools/plisttool/plisttool.py
@@ -294,15 +294,26 @@ ENTITLEMENTS_VALUE_NOT_IN_LIST = (
 
 _ENTITLEMENTS_TO_VALIDATE_WITH_PROFILE = (
     'aps-environment',
+    'com.apple.developer.applesignin',
+    'com.apple.developer.carplay-audio',
+    'com.apple.developer.carplay-charging',
+    'com.apple.developer.carplay-maps',
+    'com.apple.developer.carplay-messaging',
+    'com.apple.developer.carplay-parking',
+    'com.apple.developer.carplay-quick-ordering',
+    'com.apple.developer.playable-content',
     'com.apple.developer.networking.wifi-info',
     'com.apple.developer.passkit.pass-presentation-suppression',
     'com.apple.developer.payment-pass-provisioning',
+    'com.apple.developer.proximity-reader.payment.acceptance',
     'com.apple.developer.siri',
     'com.apple.developer.usernotifications.critical-alerts',
     'com.apple.developer.usernotifications.time-sensitive',
     # Keys which have a list of potential values in the profile, but only one in
     # the entitlements that must be in the profile's list of values
     'com.apple.developer.devicecheck.appattest-environment',
+    'com.apple.storekit.request-data',
+    'com.apple.developer.storekit.request-data',
 )
 
 ENTITLEMENTS_BETA_REPORTS_ACTIVE_MISMATCH = (
